aws-logs-metricfilter/src/main/java/software/amazon/logs/metricfilter/Translator.java

```diff
@@ -33,29 +31,61 @@ else if (e instanceof ServiceUnavailableException) {
   }
 
   static software.amazon.awssdk.services.cloudwatchlogs.model.MetricTransformation translateMetricTransformationToSdk
-          (final software.amazon.logs.metricfilter.MetricTransformation metricTransformation) {
+  (final software.amazon.logs.metricfilter.MetricTransformation metricTransformation) {
     if (metricTransformation == null) {
       return null;
     }
-    return software.amazon.awssdk.services.cloudwatchlogs.model.MetricTransformation.builder()
+
+    software.amazon.awssdk.services.cloudwatchlogs.model.MetricTransformation returnObj = software.amazon.awssdk.services.cloudwatchlogs.model.MetricTransformation.builder()
             .metricName(metricTransformation.getMetricName())
             .metricValue(metricTransformation.getMetricValue())
             .metricNamespace(metricTransformation.getMetricNamespace())
             .defaultValue(metricTransformation.getDefaultValue())
             .build();
+
+    if (metricTransformation.getMetricDimensions() != null && !metricTransformation.getMetricDimensions().isEmpty()){
+      HashMap<String, String> dimensionsMap = new HashMap<String, String>();
+      for (Dimension entry: metricTransformation.getMetricDimensions()) {
+        dimensionsMap.put(entry.getKey(), entry.getValue());
+      }
+      returnObj = returnObj.toBuilder().dimensions(dimensionsMap).build();
+    }
+
+    if (metricTransformation.getMetricUnit()!=null){
+      returnObj = returnObj.toBuilder().unit(metricTransformation.getMetricUnit()).build();
+    }
+
+    return returnObj;
   }
 
   static software.amazon.logs.metricfilter.MetricTransformation translateMetricTransformationFromSdk
-          (final software.amazon.awssdk.services.cloudwatchlogs.model.MetricTransformation metricTransformation) {
+  (final software.amazon.awssdk.services.cloudwatchlogs.model.MetricTransformation metricTransformation) {
     if (metricTransformation == null) {
       return null;
     }
-    return software.amazon.logs.metricfilter.MetricTransformation.builder()
+
+    software.amazon.logs.metricfilter.MetricTransformation returnObj = software.amazon.logs.metricfilter.MetricTransformation.builder()
```

Comment:
small np: we could probably use a better var name than `returnObj`

Reply:
agreed, updating that